### ISSUE TITLE
Document translation using gitbook's m17n feature

### ISSUE THREAD

**Issue by 5t111111** [manastech/crystal#1673]
We have been working on translation of Crystal document to Japanese and maybe it will be finished in a few weeks. 

Then I would like to offer a suggestion to publish our translations using [the m17n feature provided by gitbook](http://help.gitbook.com/format/languages.html). This means all multi-languages' files will be included in this repository, and you have to modify the directory structure to separate files for each language.

There are some drawbacks that it might mess commit history with multi-languages' ones mixed, and that you have to decide who is going to handle pull requests for other languages. Moreover, since gitbooks' m17n is not smart enough to select language to display automatically (e.g. by checking client's locale), you have to manually select a language you want. However, I think m17n files managed in one official repository via gitbook are still good for contributors who is interested in translating documents to get involved easily.

[This website](http://voltframework.com) is an example which uses gitbook's m17n (see "[Docs](http://voltframework.com/docs)").

How does it sound to you?

**Comment by adlerhsieh** [comment]
As you pointed out, syncing would be a big problem. It's hard to keep the document up-to-date for every language since they will have their own pull requests. It requires a different management tool rather than pull requests, and managers for each language syncing regularly. It would be a huge change and should be run as a separate project [as Ruby does](http://ruby-doc.org/). If the syncing problem can be solved, I think either gitbook or other tools will work.

But I agree that publishing in different languages will draw more attention and help the language progress faster.

**Comment by 5t111111** [comment]
Yes, it may not be a small change. I suppose that at least it requires extracting a branch to a separate project to manage.

**Comment by jhass** [comment]
There are translation platforms that support document oriented (opposed to key oriented) translation too, for example [WebTranslateIt](https://webtranslateit.com/en/docs/file_formats/#documents-localization)

**Comment by 5t111111** [comment]
As far as I’m concerned, regardless of using translation platforms, syncing is not a mandatory goal because, especially on an oss project, it would be hardly accomplished (or almost impossible) to keep the document for every language anytime up-to-date.

This means you have to manage just the English document as you have so far. Then if someone wants to contribute via translation, just send pull requests. In these cases, translation processes are up to each translator, then the results cannot ever be guaranteed and possibly cause even regression. I cannot imagine other possibilities for the moment though.

However, what I wanted to point out here is if you think translations of the document should be in this repository along with the original English ones. If you don't do so, we are going to publish them on our own website. Maybe we will request you [to point a subdomain to ours](https://groups.google.com/forum/?fromgroups#!topic/crystal-lang/YxvtmERLQOs).

Btw, we are using [OmegaT](http://www.omegat.org/en/omegat.html) as a translation platform to translate the documents into Japanese.

**Comment by epergo** [comment]
Can this be closed? http://ja.crystal-lang.org/docs/index.html is fully functional.
